### ISSUE TITLE
[BUG] (headed)test execution failed: "browser._newContextForReuse: Target closed"

### ISSUE THREAD

**Issue by jeroendee** [microsoft/playwright#18782]
**Context:**
- Playwright Version: 1.27.1
- Operating System: MacOS Monterey 12.6
- Node.js version: 18.11.0
- Browser: Chromium
- Extra: Latest VSCode "Playwright Test for VSCode" extension: 0.2.16 

**Code Snippet**

```typescript
import { test, expect } from '@playwright/test';

test("test (or the called website?) will 'corrupt' browser instance", async ({ page }) =>
{
    await page.goto('https://nedap-healthcare.com');
    await expect(page).toHaveTitle(/Nedap Healthcare/)
});

test('open Bing', async ({ page }) =>
{
    await page.goto('https://bing.com');
    await expect(page).toHaveTitle(/Bing/)
});

```

**Describe the bug**
TL;DR; 
After running a test that opens: "https://nedap-healthcare.com", from VSCode in headed mode, a subsequent testrun (using that same browser instance) will fail with message: "browser._newContextForReuse: Target closed"

_Prerequisites:_
- Clean install of Playwright: ```npm init playwright@latest```
- Tests are run from VSCode using the extension
- Only Chromium is selected from the Playwright profiles
- The "Show browser" checkbox is checked (i.e. the browser is run in headed mode)

_Repro:_
- Get the test code, and paste it in a file that resides within a "testDir" folder (normally: './tests'), => repro.spec.ts
- In the file editor, you should observe the test-icons left from the linenumbers
- Using the test-icon, run the test called: "test (or the called website?) will 'corrupt' browser instance" => you should see the headed Chromium browser open the configured url
- In the file editor, either run the same test again, or run the test: "open Bing" => observe the testrun failing

![2022-11-14_170442](https://user-images.githubusercontent.com/220023/201708159-dde112dc-9d8c-4675-9cc8-ef13d8ec9af9.gif)

**Comment by pavelfeldman** [comment]
It is already fixed in 1.28 that will ship this week!

**Comment by jeroendee** [comment]
@pavelfeldman sweet, and thnx for the quick triage!
